lib/rouge/lexers/squirrel.rb

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class Squirrel < RegexLexer
+      tag "squirrel"
+      filenames "*.nut"
+      mimetypes "text/x-squirrel"
+
+      title "Squirrel"
+      desc "The Squirrel programming language (squirrel-lang.org)"
+
+      def self.keywords
+        @keywords ||= %w(
+          base break case catch clone
+          continue default delete else enum
+          for foreach function if in
+          local resume return switch this
+          throw try typeof while yield constructor
+          instanceof
+        )
+      end
+
+      def self.declarations
+        @declarations ||= %w(
+          class const extends static local
+        )
+      end
+
+      def self.id
+        @id ||= %r(@?[_a-z]\w*)i
+      end
+
+      state :root do
+        rule %r(\s+)m, Text
+
+        # comments
+        rule %r((#|//).*?$), Comment::Single
+        rule %r(/[*].*?[*]/)m, Comment::Multiline
+
+        # strings
+        rule %r(@"(""|[^"])*")m, Str
+        rule %r("(\\.|.)*?["\n]), Str
+        rule %r('(\\.|.)'), Str::Char
+
+        # keywords
+        rule %r((?:#{Squirrel.keywords.join('|')})\b), Keyword
+        rule %r((?:#{Squirrel.declarations.join('|')})\b), Keyword::Declaration
+        rule %r((?:true|false|null)\b), Keyword::Constant
+        rule %r((?:class)\b), Keyword::Declaration, :class
+
+        # operators and punctuation
+        rule %r([~!%^&*()+=|\[\]{}:;,.<>\/?-]), Punctuation
+
+        rule %r([0-9]+(([.]([0-9]+)?)(e[-]?[0-9]+)?)), Num::Float
+        rule %r(0x[0-9a-fA-F]+|0[0-7]+|[0-9]+), Num::Integer
```

Comment:
The first part of this pattern should be broken out into a separate rule to match against the hexadecimal numbers. The token for that would be `Num::Hex`.

Reply:
I decided to concatenate them sine Squirrel reads Hexadecimals as Integers. But, I can separate them.

Reply:
Yeah, the names of the tokens are really a little misleading.

All languages I know (all languages maybe?) use different numeric bases only for input. Once a number is input, it's whatever it is: an integer, a float, etc. The token `Num::Integer` should really be `Num::Integer::Decimal` and `Num::Hex` should be `Num::Integer::Hex`. Alas, Pygments chose the tokens that they did.

Could you separate the hexadecimal representation pattern into a new rule with a new token? That will be consistent with the other lexers.